server/src/storage/datastore.ts

```diff
@@ -55,4 +56,37 @@ const getAllWithId = async (kind: string) => {
   return resWithId;
 };
 
-export {getWithId, getAllWithId};
+/**
+ * Unpacks id of modified object from MutationResult object.
+ */
+const getIdFromMutationResult = (
+  mutationResult: google.datastore.v1.IMutationResult
+): number | Long | string | null | undefined =>
+  mutationResult?.key?.path?.[0]?.id;
+
+/**
+ * Unpacks ids of modified objects from CommitResponse object received from Datastore.
+ */
+const getIdsFromCommitResponse = (
+  res: google.datastore.v1.ICommitResponse
+): (number | Long | string | null | undefined)[] | undefined =>
+  res.mutationResults?.map(getIdFromMutationResult);
+
+/**
+ * A Datastore wrapper that inserts a particular entity with the specified Kind.
+ * @param kind The Kind of the Entity
+ * @param data The data of the Entity to be added
+ */
+const add = async (kind: string, data: object): Promise<number> => {
+  const key = datastore.key(kind);
+  const entity = {key, data};
+  const [res] = await datastore.insert(entity);
+  const ids = getIdsFromCommitResponse(res);
+  const id = ids?.[0];
+  if (id === null || id === undefined) {
+    throw new Error(`Failed to add ${kind}.`);
+  }
+  return Number(id);
```

Comment:
What happen if id is not a number?

Reply:
It should be a number because when we call `datastore.insert(entity)` where the `key` only specifies the Kind, then the id is auto-generated by Datastore, which should be a number.